setup.py

```diff
@@ -15,35 +13,46 @@
     "xarray",
 ]
 
-test_requirements = ["pytest", "pytest-subtests", "serialbox", "coverage"]
-ndsl_requirements = ["ndsl @ git+https://github.com/NOAA-GFDL/NDSL.git@develop"]
-develop_requirements = test_requirements + ndsl_requirements + ["pre-commit"]
+test_requirements = [
+    "coverage",
+    "pytest",
+    "pytest-subtests",
+    "serialbox",
+]
+
+ndsl_requirements = ["ndsl @ git+https://github.com/NOAA-GFDL/NDSL.git@2024.09.00"]
+
+develop_requirements = [
+    *ndsl_requirements,
+    *test_requirements,
+    "pre-commit",
+]
 
 extras_requires = {
-    "test": test_requirements,
-    "ndsl": ndsl_requirements,
     "develop": develop_requirements,
+    "ndsl": ndsl_requirements,
+    "test": test_requirements,
 }
 
 setup(
-    author="The Allen Institute for Artificial Intelligence",
+    author="NOAA - Geophysical Fluid Dynamics Laboratory",
     author_email="oliver.elbert@noaa.gov",
-    python_requires=">=3.11",
+    python_requires=">=3.11,<3.12",
     classifiers=[
         "Development Status :: 2 - Pre-Alpha",
         "Intended Audience :: Developers",
-        "License :: OSI Approved :: BSD License",
+        "License :: OSI Approved :: GPLv3 License",
```

Comment:
GT4Py & DaCe are BSD - so they don't care.

It boils down to the default NOAA license. If you guys use LGPL by default we should go there probably.

Reply:
This must be the original license ascribed by AI2 as we generally use LGPLv3.  Let's leave it for now as Florian and I need to close the discussion loop with our higher ups to ensure we use a license complementary to each teams needs.

Reply:
As far as I can see, the original AI2 license is GPLv3, see [here](https://github.com/ai2cm/pace/blob/main/fv3core/LICENSE.txt).

A word of caution (I am really not a lawyer) about relicensing: LGPLv3 is less restrictive than GPLv3. You might want to seek legal advice on how to do this properly (from what I read [here](https://opensource.stackexchange.com/questions/33/how-can-a-project-be-relicensed)).

Reply:
All good then. I'll close this conversation since it looks like we aren't planning on changing the license now. If we ever get there, we can adjust this meta information again.